### DISCUSSION TITLE
how to integrate angular storybook component(has routerlink added in html element) into another angular project ?

### DISCUSSION THREAD

**Discussion by akshata456** [storybookjs/storybook#26149] [Help] [answered]
### Summary

Here I want to integrate the storybook component into the angular project. normal component integration working properly. but if I added a router link in the storybook component. then it not working. How to manage this case?
this demo "About" component was added to the main angular project.
[angular-storybook-demo.zip](https://github.com/storybookjs/storybook/files/14374922/angular-storybook-demo.zip)

**Comment by Marklb** [comment, marked as answer]
I assume you are referring to the `TestComponent`, which will not not error, but also will not work in a project that isn't Storybook, either. The project you shared, doesn't show the component used in an app or in Storybook, so when you say that it works in a normal component integration, but not in a story, I don't know what you consider a normal working component integration or how you are trying to use it in a story. If I am just not seeing it in the project you shared, then can you point out where it is used.

You have the following:
```ts
import { Component } from '@angular/core';

@Component({
  selector: 'app-test',
  standalone: true,
  imports: [],
  templateUrl: './test.component.html',
  styleUrl: './test.component.scss'
})
export class TestComponent {

}
```

```html
<p>test works!</p>
<a routerLink="/about">About</a>
```

Angular will render that without a problem. The only reason it doesn't error is that, without `[]` or `()`, Angular assumes that it is just an HTML attribute. So, if you look at the rendered element, you will see `<a routerLink="/about">...`. While it renders something that is valid code, it isn't what you want, because there should be and `href` attribute.

If you import `RouterLink` or `RouterModule` then the directive selector will be recognized and the rendered element will be `<a routerLink="/about" href="/about">...`.

```ts
import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';

@Component({
  selector: 'app-test',
  standalone: true,
  imports: [RouterLink],
  templateUrl: './test.component.html',
  styleUrl: './test.component.scss'
})
export class TestComponent {

}
```
